pkg/target/kinesis.go

```diff
@@ -127,6 +127,7 @@ func AdaptKinesisTargetFunc(f func(c *KinesisTargetConfig) (*KinesisTarget, erro
 func (kt *KinesisTarget) Write(messages []*models.Message) (*models.TargetWriteResult, error) {
 	kt.log.Debugf("Writing %d messages to stream ...", len(messages))
 
+	// TODO: Replace with new batch transformation
```

Comment:
By "replace" you mean change from passing in `messages []*models.Message` to passing in `batches []models.MessageBatch`?  I imagine the chunking logic here would need to stay still but you are just working across batches instead of messages right?

Reply:
I mean replace this bit where we call the Chunk function directly with the new function we have in this PR and pass it as a batch transformation. 

The chunking logic necessarily remains yes - the change here is just to run it as part of the new structure.